lib/train/transports/local.rb

```diff
@@ -50,8 +41,39 @@ def uri
 
       private
 
+      def select_runner(options)
+        if os.windows?
+          # Attempt to use a named pipe but fallback to ShellOut if that fails
+          begin
+            WindowsPipeRunner.new
+          rescue PipeError
+            WindowsShellRunner.new
+          end
+        else
+          GenericRunner.new(self, options)
+        end
+      end
+
+      def force_runner(runner)
+        case runner
+        when 'generic'
+          GenericRunner.new(self, options)
+        when 'windows_pipe'
+          WindowsPipeRunner.new
+        when 'windows_shell'
+          WindowsShellRunner.new
+        end
+      end
+
       def run_command_via_connection(cmd)
-        @runner.run_command(cmd)
+        if defined?(@runner)
```

Comment:
If we add the exception handling above, do we need this change? Would there ever be a case where we didn't have a valid `@runner`?

Reply:
Yes, the first few commands run do not use `GenericRunner`, but instead just use `ShellOut`.

The difference being that GenericRunner attempts to do the `CommandWrapper` bits.

Reply:
Love it. Much nicer! Will make it so.